provider/oci/oci.go

```diff
@@ -200,9 +233,19 @@ func (p *OCIProvider) addPaginatedZones(ctx context.Context, zones map[string]dn
 
 func (p *OCIProvider) newFilteredRecordOperations(endpoints []*endpoint.Endpoint, opType dns.RecordOperationOperationEnum) []dns.RecordOperation {
 	ops := []dns.RecordOperation{}
-	for _, endpoint := range endpoints {
-		if p.domainFilter.Match(endpoint.DNSName) {
-			ops = append(ops, newRecordOperation(endpoint, opType))
+	for _, ep := range endpoints {
```

Comment:
Hi. What to you think about adding a null guard here?

```
if err == nil {
  continue
}
```

Just a little improvement. There is a null crashloop somewhere, relevant issue https://github.com/kubernetes-sigs/external-dns/issues/4948

Reply:
Hi @ivankatliarchuk , I could add this if you like, but I don't think range will return a nil endpoint.
`	
         for _, ep := range endpoints {
		if ep != nil {
			continue
		}
`
otherwise I could add and error field to the provider, but that would require a major refactor for when to actually set it.

Reply:
It could.... Just have a look at method parameter `endpoints []*endpoint.Endpoint`. Agree there is no global guard. So worth to add extra line just in case.

Reply:
Sure, I'll add the check.